src/mvPlotting.cpp

```diff
@@ -3073,39 +3647,21 @@ void mvAnnotation::setPyValue(PyObject* value)
 
 void mvPlot::updateFlags()
 {
-	for (size_t i = 0; i < childslots[1].size(); i++)
+	for (auto& child : childslots[1])
 	{
-		auto child = static_cast<mvPlotAxis*>(childslots[1][i].get());
-		switch (i)
-		{
-		case(0): configData._xflags = child->configData.flags; break;
-		case(1): configData._yflags = child->configData.flags; break;
-		case(2): configData._y1flags = child->configData.flags; if (child->config.show) configData._flags |= ImPlotFlags_YAxis2; else configData._flags &= ~ImPlotFlags_YAxis2; break;
-		case(3): configData._y2flags = child->configData.flags; if (child->config.show) configData._flags |= ImPlotFlags_YAxis3; else configData._flags &= ~ImPlotFlags_YAxis3; break;
-		default: configData._yflags = child->configData.flags; break;
-		}
+		mvPlotAxis* axis = static_cast<mvPlotAxis*>(child.get());
+		configData.axesFlags[axis->configData.axis] = axis->configData.flags;
 	}
-
 }
 
 void mvPlot::updateAxesNames()
 {
-	configData.xaxisName.clear();
-	configData._y1axisName.clear();
-	configData._y2axisName.clear();
-	configData._y3axisName.clear();
+	configData.axesNames.clear();
```

Comment:
:exclamation::exclamation:  **BUG:** This piece of code accesses memory beyond `configData.axesNames` boundaries (`configData.axesNames` actually remains empty all the time, even in `draw_plot` where it is read from).

One way to resolve this would be to do `configData.axesNames.resize(ImAxis_COUNT)` in `mvPlot` constructor, and the following instead of `clear()` in `updateAxesNames()`:

```cpp
std::fill(configData.axesNames.begin(), configData.axesNames.end(), std::string());
```

However, given that the max number of axes is constant, it might be better to make `axesNames` an `std::array` rather than a `std::vector`.